modules/docextract/lib/refextract_re.py

```diff
@@ -630,7 +630,7 @@ def compute_months():
     ((\(?doi(\s)*\)?:?(\s)*)                # 'doi:' or 'doi' or '(doi)' or 'doi   )' or '(doi :' ...
     |(https?://(dx\.)?doi\.org\/))?         # or 'http://(dx.)doi.org/'  (neither has to be present)
     (?P<doi>10\.                            # 10.                        (mandatory for DOI's)
-    \d{4}                                   # [0-9] x4
+    \d{4,5}                                 # [0-9] x4 or 5
```

Comment:
DOIs seem to be way more flexible than this as the examples in https://www.doi.org/doi_handbook/2_Numbering.html#2.2.2 show. What about adopting the same pattern as the schema: https://github.com/inspirehep/inspire-schemas/blob/9b91308625e693b1785cc9308b69192da39f62f0/inspire_schemas/records/hep.yml#L774 ?

Reply:
right, `\d{4,}(?:.\d+)?` captures additional allowed forms.

it isn't even clearly spelled out that the registrant code has to be digits or that there is a minimal length of 4. I have found statements like

> The prefix identifies the registrant of the identifier, and the suffix is chosen by the registrant and identifies the specific object associated with that DOI. Most legal Unicode characters are allowed in these strings, which are interpreted in a case-insensitive manner. The prefix usually takes the form 10.NNNN, where NNNN is a series of at least 4 numbers greater than or equal to 1000, whose limit depends only on the total number of registrants

"usually" 4 digits

anyhow, I am reluctant to change legacy refextract in too much, and don't want to introduce overly broad matches. the cautious approach of looking for `\d{4,5}` covers DOIs in actual use in HEP